src/Domain/CoordinateMaps/TimeDependent/Translation.hpp

```diff
@@ -174,6 +169,22 @@ class Translation {
   friend bool operator==(  // NOLINT(readability-redundant-declaration)
       const Translation<LocalDim>& lhs, const Translation<LocalDim>& rhs);
 
+  // This helper function computes the translated coordinates or frame velocity
+  // based on the option passed in, 0 for translated coordinates, and frame
+  // velocity for any other number.
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> coord_helper(
+      const std::array<T, Dim>& source_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+          functions_of_time,
+      const double& option) const;
```

Comment:
Instead of taking a double to decide function or derivative, take the index that you want to use as a `size_t` (no reference) (e.g. `0` for function and `1` for deriv). Also can you name it something more descriptive than `option`? Maybe `function_or_deriv_index`?